### PR TITLE
Disabled back button after printing or correcting a record

### PR DESCRIPTION
**Pull request by favour234**
- save previous url on header component.
-  updated isLandingPage() method to disable back button if declaration details is updated.

### CODE REVIEW

File: packages/client/src/views/CorrectionForm/CorrectionSummary.tsx [hunk 1]
```diff
@@ -928,14 +929,18 @@ class CorrectionSummaryComponent extends React.Component<IFullProps, IState> {
...
     const declaration = this.props.declaration
     declaration.action = SubmissionAction.REQUEST_CORRECTION_DECLARATION
     declaration.submissionStatus = SUBMISSION_STATUS.READY_TO_REQUEST_CORRECTION
     updateDeclarationRegistrationWithCorrection(declaration, {
       userPrimaryOffice: this.props.userPrimaryOffice
     })
     this.props.writeDeclaration(declaration)
-    this.props.goToHomeTab(WORKQUEUE_TABS.readyForReview)
+    if (!declaration) {
+      return <Redirect to={HOME} />
```
Comment (tahmidrahman-dsi) [inline packages/client/src/views/CorrectionForm/CorrectionSummary.tsx:941]: Isn't it supposed to be inside `render` method?

File: packages/client/src/navigation/index.ts [hunk 2]
```diff
@@ -180,7 +181,9 @@ export function goToApplicationConfig() {
 }
 
 export function goToHomeTab(tabId: IWORKQUEUE_TABS, selectorId = '') {
-  return push(formatUrl(REGISTRAR_HOME_TAB, { tabId, selectorId }))
+  return push(formatUrl(REGISTRAR_HOME_TAB, { tabId, selectorId }), {
+    from: 'correction summary page'
```
Comment (tahmidrahman-dsi) [inline packages/client/src/navigation/index.ts:185]: If we doing the redirect functionality, these state work is not needed... could you please undo this?

File: packages/client/src/components/interface/Header/Header.tsx [hunk 3]
```diff
@@ -390,14 +390,19 @@ class HeaderComp extends React.Component<IFullProps, IState> {
   isLandingPage = () => {
     const role = this.props.userDetails && this.props.userDetails.role
     const location = this.props.history.location.pathname
+
+    const prevUrl = this.props.location.state
+    console.log(prevUrl)
```
Comment (tahmidrahman-dsi) [inline packages/client/src/components/interface/Header/Header.tsx:395]: Could you please remove `console.log`

**Review by tahmidrahman-dsi — CHANGES_REQUESTED** [review]

File: packages/client/src/navigation/index.ts (hunk 2 above)
Comment (favour234) [reply packages/client/src/navigation/index.ts:185]: alright.

**Pull request closed without merging**